system-contract-dapp-playground/src/api/hedera/tokenCreateCustom-interactions/index.ts

```diff
@@ -239,8 +239,258 @@ export const createHederaNonFungibleToken = async (
     const tokenAddress = `0x${data.slice(-40)}`;
 
     return { tokenAddress, transactionHash: txReceipt.hash };
-  } catch (err) {
+  } catch (err: any) {
     console.error(err);
-    return { err };
+    return { err, transactionHash: err.receipt && err.receipt.hash };
+  }
+};
+
+/**
+ * @dev mints Hedera tokens
+ *
+ * @dev integrates tokenCreateCustomContract.mintTokenPublic()
+ *
+ * @param baseContract: ethers.Contract
+ *
+ * @param tokenType: 'FUNGIBLE' | 'NON_FUNGIBLE'
+ *
+ * @param hederaTokenAddress: string
+ *
+ * @param amountToMint: number
+ *
+ * @param metadata: string[]
+ *
+ * @return Promise<TokenCreateCustomSmartContractResult>
+ */
+export const mintHederaToken = async (
```

Comment:
nit: I feel like these should all be in there own files and brought together in index.ts but that can be done later